### PR TITLE
Add test cases for the requirements of "gc-aot" feature

### PR DESCRIPTION
**Pull request by yzha107**
Add test cases for the requirements of "gc-aot" feature

### CODE REVIEW

File: tests/requirement-engineering/gc-aot/run.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+
+import os
+import argparse
+from typing import List, Dict, Tuple
+import json
+
+WORK_DIR = os.getcwd()
+WAMR_DIR = os.path.join(WORK_DIR, "../../../wamr")
```
Comment (wenyongh) [inline tests/requirement-engineering/gc-aot/run.py:13]: Should be "../../.."?

File: tests/requirement-engineering/gc-aot/run.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+IWASM_CMD = os.path.join(
+    WORK_DIR, "../../../wamr/product-mini/platforms/linux/build/iwasm")
+WAMRC_CMD = os.path.join(WORK_DIR, "../../../wamr/wamr-compiler/build/wamrc")
```
Comment (wenyongh) [inline tests/requirement-engineering/gc-aot/run.py:16]: Same as above, remove "/wamr" in L14 and L16

File: tests/wamr-test-suites/test_wamr.sh [hunk 3]
```diff
@@ -218,7 +241,8 @@ readonly DATE=$(date +%Y-%m-%d_%H:%M:%S)
 readonly REPORT_DIR=${WORK_DIR}/report/${DATE}
 mkdir -p ${REPORT_DIR}
 
-readonly WAMR_DIR=${WORK_DIR}/../../..
+readonly WAMR_DIR=${WORK_DIR}/../../../wamr
```
Comment (wenyongh) [inline tests/wamr-test-suites/test_wamr.sh:244]: Should remove `/wamr`

**Review by wenyongh — APPROVED** [review]
LGTM

File: tests/wamr-test-suites/test_wamr.sh [hunk 4]
```diff
@@ -14,7 +14,9 @@ function help()
 {
     echo "test_wamr.sh [options]"
     echo "-c clean previous test results, not start test"
-    echo "-s {suite_name} test only one suite (spec|wasi_certification|wamr_compiler)"
+    echo "-s {suite_name} test only one suite (spec|standalone|sightglass|unit"
+    echo "                                     |malformed|wasi|wasi_certification"
```
Comment (TianlongLiang) [inline tests/wamr-test-suites/test_wamr.sh:18]: better only add the test suite that is available, for example, unit and standalone are unavailable, better not add them. I think maybe keep the original choice is good enough

File: tests/wamr-test-suites/test_wamr.sh [hunk 5]
```diff
@@ -1044,7 +1078,7 @@ if [[ $TEST_CASE_ARR ]];then
     trigger || (echo "TEST FAILED"; exit 1)
 else
     # test all suite, ignore polybench and libsodium because of long time cost
-    TEST_CASE_ARR=("spec")
+    TEST_CASE_ARR=("spec" "malformed" "wasi" "sightglass" "standalone")
```
Comment (TianlongLiang) [inline tests/wamr-test-suites/test_wamr.sh:1081]: Similarly, maybe it's best to keep the original set, some of the added ones are unavailable

**Comment by TianlongLiang** [comment]
LGTM

**Pull request merged**